### PR TITLE
Add KL divergence test for D.OneHotCategorical

### PR DESCRIPTION
**Pull request by takagi**
This PR follows up #5372 to add KL divergence test for `D.OneHotCategorical` .

### CODE REVIEW

**Comment by takagi** [comment]
Jenkins, test this please.

File: tests/chainer_tests/distributions_tests/test_kldivergence.py [hunk 1]
```diff
@@ -261,6 +268,18 @@ def test_multivariatenormal_multivariatenormal_gpu(self):
         dist2 = self.make_multivariatenormal_dist(True)
         self.check_kl(dist1, dist2)
 
+    @testing.with_requires('numpy>=1.11')
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_kldivergence.py]: Why is it required?

Comment (toslunar) [reply tests/chainer_tests/distributions_tests/test_kldivergence.py]: Would you review #5604 first?

Comment (takagi) [reply tests/chainer_tests/distributions_tests/test_kldivergence.py]: Sure!

Comment (takagi) [reply tests/chainer_tests/distributions_tests/test_kldivergence.py]: I removed this `with_requires`

**Comment by beam2d** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
This PR adds KL divergence of `OneHotCategorical`s.

**Comment by beam2d** [comment]
Failed tests are irrelevant to this PR. LGTM. @toslunar Could you merge it if you are ok?

**Pull request merged**